src/github.com/matrix-org/dendrite/clientapi/writers/createroom.go

```diff
@@ -34,9 +75,47 @@ func CreateRoom(req *http.Request) util.JSONResponse {
 	if resErr != nil {
 		return *resErr
 	}
+	// TODO: apply rate-limit
+
+	if resErr = r.Validate(); resErr != nil {
+		return *resErr
+	}
+
+	// TODO: visibility/presets/raw initial state/creation content
+
+	hostname := "localhost"
```

Comment:
This should probably be called `serverName` rather than `hostname` for consistency.